### PR TITLE
feat(docs): Updated examples topic to use ECS L2 construct

### PR DESCRIPTION
**Pull request by Doug-AWS**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

File: docs/src/examples.rst [hunk 1]
```diff
@@ -14,205 +14,175 @@
...
+This topic contains some examples to help you get started using some of the advanced constructs
+offered by the |cdk|.
 
-.. We'll include this if we ever implement DeploymentPipeline
-   _multiple_stacks_example:
+.. _creating_ecs_l2_example:
 
-   Creating an App with Multiple Stacks
-   ====================================
+Creating an |ECS| Construct
```
Comment (rix0rrr) [inline docs/src/examples.rst]: I wouldn't call it "creating an ECS *construct*", that implies you're building a reusable thing.

Maybe "Writing an |ECS| CDK app" ?

File: docs/src/examples.rst [hunk 2]
```diff
@@ -14,205 +14,175 @@
...
-   - **EU** in the **eu-west-1** Region
-   - **DeploymentPipeline** in the **us-east-1** Region
+Since |ECS| can be used with a number of AWS services,
+you should understand how the |ECS| construct that we use in this example
+gives you a leg up on using AWS services:
 
-   Implement the class **MyStack** in the *my-stack* sub-folder,
-   that extends the |stack-class| class
-   (this is the same code as shown in the :doc:`concepts` topic).
+* Automatic security group opening for load balancers
```
Comment (rix0rrr) [inline docs/src/examples.rst]: These are not sentences, maybe rephrase to a list of the form "it will open up ..." etc.

File: docs/src/examples.rst [hunk 3]
```diff
@@ -14,205 +14,175 @@
...
+  thus saving you deployment time
+* Automatically adds permissions for |ECR| if you use an image from |ECR|
+* Convenient API for autoscaling
+* Asset support, so that you can deploy source from your machine to |ECS| in one step
 
-   code-block:: js
+.. _creating_ecs_l2_example_1:
 
-   import { Stack, StackProps } from '@aws-cdk/cdk'
+Step 1: Create the Directory and Initialze the |cdk|
```
Comment (rix0rrr) [inline docs/src/examples.rst]: Typo. Wouldn't capitalize Directory and Initialize here.

File: docs/src/examples.rst [hunk 4]
```diff
@@ -14,205 +14,175 @@
...
+.. _creating_ecs_l2_example_3:
 
-.. code-block:: js
+Step 3: Create a Fargate Service
+--------------------------------
 
-   import ec2 = require('@aws-cdk/aws-ec2');
-   import rds = require('@aws-cdk/aws-rds');
-   import cdk = require('@aws-cdk/cdk');
+There are two different ways of creating a serverless infrastructure with |ECS|:
```
Comment (rix0rrr) [inline docs/src/examples.rst]: I would write "two different ways of running your container tasks"

File: docs/src/examples.rst [hunk 5]
```diff
@@ -14,205 +14,175 @@
...
-   class MyStack extends cdk.Stack {
-     constructor(parent: cdk.App, name: string, props?: cdk.StackProps) {
-       super(parent, name, props);
+- Using the **Fargate** launch type, where |ECS| manages your cluster resources
+- Using the **EC2** launch type, where you manage your cluster resources
 
-       const vpc = new ec2.VpcNetwork(this, 'VPC');
+This example creates a Fargate service,
+which requires a VPC, a cluster, a task definition, and a security group.
+Later on we'll show you how to launch an EC2 service.
```
Comment (rix0rrr) [inline docs/src/examples.rst]: how to launch services on EC2 instances you manage yourself

File: docs/src/examples.rst [hunk 6]
```diff
@@ -14,205 +14,175 @@
...
-=======================
+        Add the following code to the end of the constructor:
 
-The following example creates the |S3| bucket **MyS3Bucket** with server-side KMS
-encryption provided by |S3|.
+        .. code-block:: typescript
 
-.. code-block:: js
+            const vpc = new ec2.VpcNetwork(this, 'MyVpc', {
+              maxAZs: 2 // Default is all AZs in region
```
Comment (rix0rrr) [inline docs/src/examples.rst]: 3 is actually a better default if we're going to teach people best practices. 2 is just for tests.

File: docs/src/examples.rst [hunk 7]
```diff
@@ -14,205 +14,175 @@
...
+            });
 
-       new s3.Bucket(this, 'MyBucket', {
-         bucketName: 'MyS3Bucket',
-         encryption: s3.BucketEncryption.KmsManaged
-       });
-     }
-   }
+            // The task definition must have at least one container.
+            // Otherwise you cannot synthesize or deploy your app.
```
Comment (rix0rrr) [inline docs/src/examples.rst]: It sounds like a constraint if you put it like this. But also, if you don't add a container, WHAT are you even launching? Empty tasks that do nothing? That's not very useful.

So I wouldn't call attention to it in this way.

File: docs/src/examples.rst [hunk 8]
```diff
@@ -14,205 +14,175 @@
...
+            taskDefinition.addContainer('MyContainer', {
+              image: ecs.ContainerImage.fromDockerHub('MyDockerImage')    // Required
```
Comment (rix0rrr) [inline docs/src/examples.rst]: Maybe launch `amazon/amazon-ecs-sample`, that's a container that at least does something.

File: docs/src/examples.rst [hunk 9]
```diff
@@ -14,205 +14,175 @@
...
+            const securityGroup = new ec2.SecurityGroup(this, 'MySecurityGroup', {
+              vpc: vpc
+            });
 
-   new MyStack(app, 'MyStack');
+            new ecs.FargateService(this, 'MyFargateService', {
+              taskDefinition: taskDefinition,  // Required
+              cluster: cluster,  // Required
+              desiredCount: 6,  // Default is 1
+              securityGroup: securityGroup,  // Required
```
Comment (rix0rrr) [inline docs/src/examples.rst]: No it's not.

File: docs/src/examples.rst [hunk 10]
```diff
@@ -14,205 +14,175 @@
...
-     }
-   }
+            // The task definition must have at least one container.
+            // Otherwise you cannot synthesize or deploy your app.
+            taskDefinition.addContainer('MyContainer', {
+              image: ecs.ContainerImage.fromDockerHub('MyDockerImage')    // Required
+            });
 
-   const app = new cdk.App();
+            const securityGroup = new ec2.SecurityGroup(this, 'MySecurityGroup', {
```
Comment (rix0rrr) [inline docs/src/examples.rst]: This is actually not necessary.

File: docs/src/examples.rst [hunk 11]
```diff
@@ -14,205 +14,175 @@
...
+              vpc: vpc
+            });
 
-   new MyStack(app, 'MyStack');
+            new ecs.FargateService(this, 'MyFargateService', {
+              taskDefinition: taskDefinition,  // Required
+              cluster: cluster,  // Required
+              desiredCount: 6,  // Default is 1
+              securityGroup: securityGroup,  // Required
+              platformVersion: ecs.FargatePlatformVersion.Latest  // Default
```
Comment (rix0rrr) [inline docs/src/examples.rst]: Don't even call attention to this, just leave it out.

File: docs/src/examples.rst (hunk 4 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: ok

File: docs/src/examples.rst (hunk 5 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: ok

File: docs/src/examples.rst (hunk 6 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: ok

File: docs/src/examples.rst (hunk 7 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: I rewrote the first comment line and removed the second comment line.

File: docs/src/examples.rst (hunk 10 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: removed

File: docs/src/examples.rst (hunk 9 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: yep

File: docs/src/examples.rst (hunk 11 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: done

**Comment by Doug-AWS** [comment]
I've updated the list of benefits and am deploying the code now to make sure it works (40 minutes and counting):

 28/30 | 11:50:59 | CREATE_COMPLETE      | AWS::EC2::Route                       | MyVpc/PrivateSubnet2/DefaultRoute (MyVpcPrivateSubnet2DefaultRoute9CE96294)
28/30 Currently in progress: MyFargateService8825BC17

**Comment by Doug-AWS** [comment]
Strange. I nuked my deployment and the CFN stack. then started all over with a fresh project. The code deployed in about 3 1/2 minutes. The template grew from about 300 lines to almost 500.

File: docs/src/examples.rst (hunk 3 above)
Comment (rix0rrr) [reply docs/src/examples.rst]: Still typo

File: docs/src/examples.rst [hunk 12]
```diff
@@ -14,205 +14,265 @@
...
-      preProdStages: [ preProd ],
-      prodStages: prod
-   });
+        .. code-block:: sh
 
-   app.run();
+            cdk init --language typescript
 
-.. _dynamodb_example:
+        Update *my_ecs_construct.ts* in the *bin* directory to only contain the following code:
```
Comment (rix0rrr) [inline docs/src/examples.rst]: This will have gotten changed to `lib/my_ecs_construct_stack.ts`  by the time this goes out.

File: docs/src/examples.rst [hunk 13]
```diff
@@ -14,205 +14,265 @@
...
+    .. group-tab:: TypeScript
+
+        Add the following import statements:
+
+        .. code-block:: typescript
+
+            .. code-block:: ts
+
+                // Add capacity to it
+                cluster.addDefaultAutoScalingGroupCapacity({
```
Comment (rix0rrr) [inline docs/src/examples.rst]: Sorry, no. This is to add EC2 capacity to run Ec2Services on. You've been creating a FargateService so you don't need to do this.

You want what this section says:

https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs/README.md#task-autoscaling

File: docs/src/examples.rst (hunk 12 above)
Comment (rix0rrr) [reply docs/src/examples.rst]: I know. The change is not released yet. That's why I wrote:

> by the time this goes out.

File: docs/src/examples.rst [hunk 14]
```diff
@@ -14,205 +14,216 @@
...
-       super(parent, name, props);
+              }
+            }
 
-       const table = new dynamodb.Table(this, 'Table', {
-         tableName: 'MyAppTable',
-         readCapacity: 5,
-         writeCapacity: 5
-       });
+            const app = new cdk.App();
```
Comment (rix0rrr) [inline docs/src/examples.rst]: In fact these lines will also be gone.

These lines will live in a file in the `bin` directory, the other lines will live in a file in the `lib` directory.

Here's the new version of the *app* template: https://github.com/awslabs/aws-cdk/tree/master/packages/aws-cdk/lib/init-templates/app/typescript

File: docs/src/examples.rst [hunk 15]
```diff
@@ -14,205 +14,216 @@
...
+        Deploy the stack.
+
+        .. code-block:: sh
+
+            cdk deploy
+
+        |CFN| displays information about the dozens of steps that
+        it takes as it deploys your app.
+
+That's how easy it is to create a Fargate service to run a Docker image.
```
Comment (rix0rrr) [inline docs/src/examples.rst]: It's true. The example is not yet useful without a load balancer though, since this is a web service but you will be unable to hit the endpoints (because they'll be in a private VPC if I'm not mistaken).

Just sayin'.

Comment (rix0rrr) [reply docs/src/examples.rst]: Are you planning to make this section longer, or is it the plan to publish like this?

Comment (Doug-AWS) [reply docs/src/examples.rst]: I'll add a load balancer and auto-scaling for tasks to the example. Once the new template is released, I'll fix the references:

*my_ecs_construct.ts* in the *bin* directory -> *my_ecs_stack.ts* in the *lib* directory

File: docs/src/examples.rst [hunk 16]
```diff
@@ -14,205 +14,216 @@
...
+            const vpc = new ec2.VpcNetwork(this, 'MyVpc', {
+              maxAZs: 3 // Default is all AZs in region
+            });
+
+            const cluster = new ecs.Cluster(this, 'MyCluster', {
+              vpc: vpc
+            });
+
+            // Add capacity to it
+            cluster.addDefaultAutoScalingGroupCapacity({
```
Comment (rix0rrr) [inline docs/src/examples.rst]: Don't need capacity if you're starting a Fargate task

Comment (Doug-AWS) [reply docs/src/examples.rst]: Right. I saw your comment to Paul in issue #1279.

File: docs/src/examples.rst (hunk 12 above)
Comment (Doug-AWS) [reply docs/src/examples.rst]: I removed this part as the template has changed.

**Comment by Doug-AWS** [comment]
I'm still blocked from confirming **cdk deploy** by issue #1279

File: docs/src/examples.rst [hunk 17]
```diff
@@ -65,7 +67,7 @@ gives you a leg up on using AWS services by providing the following benefits:
 * Automatically adds permissions for |ECR| if you use an image from |ECR|
   When you use an image from |ECR|, the |cdk| adds the correct permissions.
 
-* Convenient API for autoscaling
+* Automatic autoscaling
```
Comment (rix0rrr) [inline docs/src/examples.rst]: What does this mean? It's not automatic yet. I think we may have miscommunicated

Comment (Doug-AWS) [reply docs/src/examples.rst]: Okay, so what should it say? Should I go back to:

* Convenient API for autoscaling
  The |cdk| supplies a method so you can autoscale instances when you use an |EC2| cluster;
  this functionality is done automatically when you use an instance in a Fargate cluster.
